spec/controllers/hq/partners_controller_spec.rb

```diff
@@ -26,20 +26,18 @@ def foobar_action
   end
 
   context '#new and #create' do
-    let(:partner_double) { instance_double Partner, :attributes= => true }
-    
-    before :each do
-      controller.instance_variable_set(:@partner, partner_double)
-    end
-    
+
     specify 'successful create redirects to the show view' do
-      expect(partner_double).to receive(:save).and_return(true)
+      allow_any_instance_of(Partner).to receive(:save) do |_self|
+        _self.id = 1
+        true        
```

Comment:
because `save` returns `true` when it successfully saves (sets an id to the current partner)
The returned `id` would also be truthy but it's not normal behavior of `save`